libretroshare/src/pqi/p3peermgr.cc

```diff
@@ -1101,6 +1087,72 @@ bool p3PeerMgrIMPL::removeFriend(const RsPgpId &id)
 
         return !sslid_toRemove.empty();
 }
+
+bool p3PeerMgrIMPL::addFriendPendingPgp(
+        const RsPeerId& sslId, const RsPeerDetails& dt )
+{
+	if(sslId.isNull() || sslId == getOwnId()) return false;
+
+	bool notifyLinkMgr = false;
+
+	{
+		RS_STACK_MUTEX(mPeerMtx);
+
+		peerState pstate;
+
+		/* If in mOthersList -> move over */
+		auto it = mOthersList.find(sslId);
+		if (it != mOthersList.end())
+		{
+			pstate = it->second;
+			mOthersList.erase(it);
+		}
+
+		pstate.id = sslId;
+		if(!dt.gpg_id.isNull())  pstate.gpg_id = dt.gpg_id;
```

Comment:
I dont like this "optional" format fields. Either the protocol needs the FPR, or it doesn't. And when we write and review the security of all this (which is not done yet) we'll have a much simpler and safer task if there are not such options because they have a lot of implications. I would say that both system (with or without FPR) are possible, but havign both of them at once is not nice at all.

Reply:
Ok in this case, as the idea behind this to have very short invites (we already have the long ones in place), that other application might easily recognize as something the user might want to click on (if they are not clickable they are practically unusable in the mobile phones, I can make concrete use case examples in favor of making them something that should be clickable). I'll remove the PGP fingerprint.